src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -428,8 +428,18 @@ private List<string> GetWorkerDiagLogFiles(string diagFolder, DateTime jobStartT
             return workerLogFiles;
         }
 
-        private List<string> GetAgentDiagLogFiles(string diagFolder, DateTime jobStartTimeUtc)
+        private List<string> GetAgentDiagLogFiles(DateTime jobStartTimeUtc)
         {
+            // Since this is the Worker context, we still want to fetch the Agent _diag folder.
+            // This might have been overriden by the envrionment variable.
+            // TODO: this is a hacky way to do it, maybe it should be a config value
+            var diagFolder = Environment.GetEnvironmentVariable($"AGENT_DIAGLOGPATH");
```

Comment:
Could you please use the already existing mechanism of agent knobs instead?
You can find example of declaration [here](https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Agent.Sdk/Knob/AgentKnobs.cs#L189) and usage [here](https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Microsoft.VisualStudio.Services.Agent/VstsAgentWebProxy.cs#L195).

Reply:
Updated to use knobs.